sdk/core/azure-core/azure/core/polling/async_base_polling.py

```diff
@@ -152,12 +154,26 @@ async def request_status(self, status_link: str) -> PipelineResponse[HttpRequest
         """
         if self._path_format_arguments:
             status_link = self._client.format_url(status_link, **self._path_format_arguments)
+        request_params: Dict[str, Any] = {}
+        if self._lro_options and "api_version" in self._lro_options:
+            parsed_status_link = urllib.parse.urlparse(status_link)
+            request_params = cast(
+                Dict[str, Any],
+                case_insensitive_dict(
```

Comment:
I think if you update `request_params` with the new dict, you can get rid of the `cast`

Reply:
Updated